testing/internal/e2e/tests/base_plus/authenticate_no_keyring_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package base_plus_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCliAuthenticateNoKeyring tests authentication when we're using `-format
+// json` and there is no keyring on the system.
+// This covers a case where the client cache daemon should intercept the auth
+// token in the json response. There was previously a bug where the client cache
+// daemon tried to access the keyring when there wasn't one on the system.
+func TestCliAuthenticateNoKeyring(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	ctx := context.Background()
+
+	bc, err := boundary.LoadConfig()
+	require.NoError(t, err)
+
+	containerID := "boundary"
+
+	docker, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
```

Comment:
It looks like we're using https://github.com/ory/dockertest in other places, maybe we can use it here too? Not sure if it's suitable, feel free to say no.

Reply:
I'll double-check. When I first looked, it didn't seem like there was a way to connect to an existing docker image using `dockertest`.

Reply:
Ah you're probably right!